pkg/controller/windowsmachineconfig/status.go

```diff
@@ -47,10 +47,13 @@ func (s *StatusManager) updateStatus() error {
 	if err != nil {
 		return errors.Wrapf(err, "could not get %v", s.wmcName)
 	}
-
-	object.Status.JoinedVMCount = s.joinedVMCount
-	for _, condition := range s.conditionsToSet {
-		object.Status.SetWindowsMachineConfigCondition(condition)
+	if s.joinedVMCount != 0 {
+		object.Status.JoinedVMCount = s.joinedVMCount
+	}
```

Comment:
Can't the `joinedVMCount` be 0 on a scale down operation?

Reply:
missed this scenario, updating the joinedVmCount with current value while reconciling the status, in order to avoid setting incorrect values.